### PR TITLE
Features/ui archive job spec

### PR DESCRIPTION
**Pull request by NavyAdmiral**
gif

- Adds a new Archive button in the job regional nav
- Clicking archive brings up a confirmation modal
- After confirming user is redirected to the index dashboard
- A success flash is displayed to notify about the archived job
- The associated jobruns for the archived jobspec should disappear from the index dashboard (done in reducers)
- The archived jobrun should also disappear in the index dashboard (done in reducers)

### CODE REVIEW

**Comment by NavyAdmiral** [comment]
@rupurt do you suggest I filter jobs and associated runs when the user Archives a job, or should I just directly refetch jobs and runs? https://github.com/smartcontractkit/chainlink/pull/1401/commits/df0bd82d70abb159ee8bf4097013760f12273134 will clean jobs, but associated runs will remain visible in the ui, unless I add a change in runs reducers, or refetch altogether

**Comment by rupurt** [comment]
@NavyAdmiral do you mean in the redux state? I feel like we should delete those.

File: operator_ui/__tests__/connectors/reducers/jobRuns.test.js [hunk 1]
```diff
@@ -93,4 +94,24 @@ describe('connectors/reducers/jobRuns', () => {
       b: { id: 'b' }
     })
   })
+  it('RECEIVE_DELETE_SUCCESS deletes jobrun associations', () => {
```
Comment (rupurt) [inline operator_ui/__tests__/connectors/reducers/jobRuns.test.js]: ✂️ this no longer exists

File: operator_ui/__tests__/connectors/reducers/jobs.test.js [hunk 2]
```diff
@@ -75,12 +76,30 @@ describe('connectors/reducers/jobs', () => {
...
+      }
+    }
+    const preDeleteState = reducer(undefined, upsertAction)
+
+    expect(preDeleteState.jobs.items).toEqual({ a: { id: 'a' } })
+    const deleteAction = {
+      type: RECEIVE_DELETE_SUCCESS,
+      data: 'a'
+    }
+    const postDeleteState = reducer(undefined, deleteAction)
```
Comment (rupurt) [inline operator_ui/__tests__/connectors/reducers/jobs.test.js]: This should include an assertion that `a` is no longer in `items`

File: operator_ui/src/connectors/redux/reducers/jobRuns.js [hunk 3]
```diff
@@ -29,6 +31,16 @@ export default (state = initialState, action = {}) => {
         items: Object.assign({}, state.items, action.data.runs)
       })
     }
+    case RECEIVE_DELETE_SUCCESS: {
+      return Object.assign({}, state, {
+        items: Object.assign(
+          {},
+          Object.keys(state.items)
+            .filter(i => state.items[i].attributes.jobId !== action.response)
+            .reduce((res, key) => ((res[key] = state.items[key]), res), {})
```
Comment (rupurt) [inline operator_ui/src/connectors/redux/reducers/jobRuns.js]: How do you feel about using lodash [pick](https://lodash.com/docs/4.17.11#pick) here?

File: operator_ui/src/connectors/redux/reducers/jobs.js [hunk 4]
```diff
@@ -38,6 +40,16 @@ export default (state = initialState, action = {}) => {
         items: Object.assign({}, state.items, action.data.specs)
       })
     }
+    case RECEIVE_DELETE_SUCCESS: {
+      return Object.assign({}, state, {
+        items: Object.assign(
+          {},
+          Object.keys(state.items)
+            .filter(i => i !== action.response)
+            .reduce((res, key) => ((res[key] = state.items[key]), res), {})
```
Comment (rupurt) [inline operator_ui/src/connectors/redux/reducers/jobs.js]: How do you feel about using lodash [pick](https://lodash.com/docs/4.17.11#pick) here?

File: operator_ui/src/connectors/redux/reducers/jobRuns.js (hunk 3 above)
Comment (NavyAdmiral) [reply operator_ui/src/connectors/redux/reducers/jobRuns.js]: @rupurt I like your suggestion

File: operator_ui/__tests__/connectors/reducers/jobs.test.js (hunk 2 above)
Comment (NavyAdmiral) [reply operator_ui/__tests__/connectors/reducers/jobs.test.js]: @rupurt slipped by :P

File: operator_ui/__tests__/connectors/reducers/jobRuns.test.js (hunk 1 above)
Comment (NavyAdmiral) [reply operator_ui/__tests__/connectors/reducers/jobRuns.test.js]: @rupurt Not sure what you mean. Can you elaborate on this one?

Comment (rupurt) [reply operator_ui/__tests__/connectors/reducers/jobRuns.test.js]: `runs` should no longer be returned from the jobs#show API endpoint

**Comment by rupurt** [comment]
@NavyAdmiral can you rebase instead of merging develop into this branch so that we can keep the history clean?

**Comment by NavyAdmiral** [comment]
@rupurt I rebased. Aside the modal design, are there parts of the code that you would like to be handled differently?

**Comment by rupurt** [comment]
Cheers @NavyAdmiral 🍻 

I'm fine with the modal design.

**Pull request merged**